kimchi/src/circuits/lookup/lookups.rs

```diff
@@ -263,6 +263,85 @@ impl<F: Copy> JointLookup<SingleLookup<F>, LookupTableID> {
     }
 }
 
+pub enum LookupPattern {
+    ChaCha,
+    ChaChaFinal,
+    LookupGate,
+}
+
+impl LookupPattern {
+    pub fn lookups<F: Field>(&self) -> Vec<JointLookupSpec<F>> {
```

Comment:
It might be nicer to add the lookup pattern to the `Argument` interface, so that the lookup pattern definition can be specified on the definition of the respective circuitgates.  For example, `Argument::lookup_pattern()` could either return a pattern or None if the circuit gate doesn't use any plookups.

Reply:
agree :o but this is a step in the right direction!